### DISCUSSION TITLE
Separation of Concerns in RetrieveUserProxyAgent Class

### DISCUSSION THREAD

**Discussion by Naitik-k** [microsoft/autogen#3955] [Q&A]
_Description_:
Currently, in the **RetrieveUserProxyAgent** class, both the ingestion and retrieval functionalities are handled within the same class. This makes it difficult to maintain separation of concerns.

_Question_:
Is there a way or a recommended pattern to separate these two responsibilities into different classes or components? If not, would it be possible to enhance the design to allow for this separation?

**Comment by nileshm11** [comment]
Yes @Naitik-k 
I have faced same issue i know they have done this for ease for developers but they should give this flexibility also
